api/osrd_infra/views/train_schedule/standalone_simulation_report.py

```diff
@@ -150,6 +156,37 @@ def compute_tail_positions(head_positions, train_length: float):
     return results
 
 
+def project_signal_updates(signal_updates, projection_path_payload: PathPayload, departure_time):
+    results = []
+
+    updates_by_route_id = defaultdict(lambda: list())
+    for update in signal_updates:
+        for route_id in update["route_ids"]:
+            updates_by_route_id[route_id].append(update)
```

Comment:
By principle to avoid doing too many operations in the middle, we could change the format of `signal_updates` so we already have `updates_by_route_id` dictionary built.

Reply:
By having the back-end response look like `{route_id: update}`? That would make the response extremely redundant, there can be a lot of routes updated for each signal aspect.